### PR TITLE
Add codeql-analyzer to CI

### PR DESCRIPTION
**Pull request by measure-fi**
https://github.com/palomachain/paloma/issues/526

## Background

This analyzer may be a little noisy initially, but will hopefully tell us where new issues are creeping in. Going to silence the remainder of the warnings and merge it (for now).

### CODE REVIEW

**Comment by taariq** [comment]
I love curiosity! Let's be curious!

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -1,7 +1,4 @@
 on:
-  push:
-    branches:
-      - '*'
```
Comment (measure-fi) [inline .github/workflows/ci-test.yml:4]: @Vizualni Without this it's just running CI twice against every PR. If you want CI running on all branches I would turn it off for PRs.

**Comment by taariq** [comment]
Closing for now.

**Comment by measure-fi** [comment]
@taariq One of these errors did line up with the fixes last week. I think the payoff here is rather large: it avoids a chain halt. I'd budget 1.5 days of my time to fix all of these and then leave it on in CI to audit future work. When exactly that happens is a matter of prioritization, but it should happen.

**Comment by taariq** [comment]
@measure-fi re-opening!

**Comment by measure-fi** [comment]
I think the floating point error is _actually_ bad, so I'm leaving it in for now. Will need more thought on that one.

**Pull request merged**